src/task.py

```diff
@@ -0,0 +1,233 @@
+
+    def _parse_task_id(self, output: Union[str, bytes]) -> Optional[str]:
+        text = output.decode("utf-8") if isinstance(output, bytes) is True else output
+        task_id_regex = re.compile(r"(?<=ID: ).*", flags=re.IGNORECASE)
+        try:
+            matches = task_id_regex.search(text)
+            if matches is None:
+                return None
+            return matches[0]
+        except TypeError:
+            return None
+
+    def _cp_storage_to_ee_image(self, blob_uri: str, image_asset_id: str) -> str:
+        try:
+            cmd = [
+                "/usr/local/bin/earthengine",
+                f"--service_account_file {self.google_creds_path}",
+                "upload image",
+                "--nodata_value=0",
+                f"--asset_id={image_asset_id}",
+                blob_uri,
+            ]
+            output = subprocess.check_output(
+                " ".join(cmd), stderr=subprocess.STDOUT, shell=True
+            )
+            task_id = self._parse_task_id(output)
+            if task_id is None:
+                raise TypeError("task_id is None")
+            return task_id
+        except subprocess.CalledProcessError as err:
+            raise ConversionException(err.stdout)
+    
+    def _cp_storage_to_ee_table(self, blob_uri: str, table_asset_id: str) -> str:
+        try:
+            cmd = [
+                "/usr/local/bin/earthengine",
+                f"--service_account_file {self.google_creds_path}",
+                "upload table",
+                "--primary_geometry_column wkt",
+                f"--asset_id={table_asset_id}",
+                blob_uri,
+            ]
+            output = subprocess.check_output(
+                " ".join(cmd), stderr=subprocess.STDOUT, shell=True
+            )
+            task_id = self._parse_task_id(output)
+            if task_id is None:
+                raise TypeError("task_id is None")
+            return task_id
+        except subprocess.CalledProcessError as err:
+            raise ConversionException(err.stdout)
+
+    def _get_image_asset_id(self, attribute: str, tag: str, task_date: str):
+        return f"{self.ee_osm_root}/{attribute}/{tag}/{tag}_{task_date}"
+
+    # Step 1
+    def import_images_to_ee(
+        self, metadata: dict, image_asset_id: Optional[str] = None
+    ) -> List[str]:
+        
+        image_uris = metadata.get("images") or []
+        if not image_uris:
+            return []
+        
+        ee_dir = f"{self.ee_osm_root}/{self.taskdate}"
+        self._prep_asset_id(ee_dir, image_collection=False)
+
+        image_asset_ids = []
+        for image_uri in image_uris:
+            image_asset_id = f"{PROJECTS}/{self.ee_project}/{ee_dir}/{Path(os.path.splitext(image_uri)[0]).name}"
+            task_id = self._cp_storage_to_ee_image(image_uri, image_asset_id)
+            self.ee_tasks[task_id] = {}
+            image_asset_ids.append(image_asset_id)
+
+        self.wait()
```

Comment:
Is this necessary anymore?

Reply:
that one is still necessary to wait until all the split images are loaded before grouping them